packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -338,6 +338,10 @@ export const useGridSelection = (
         gridClasses.cell,
       );
       const field = cellClicked?.getAttribute('data-field');
+      if (field === '__check__') {
```

Comment:
If we change the field on day we don't need to update here.

```suggestion
      if (field === GRID_CHECKBOX_SELECTION_COL_DEF.field) {
```